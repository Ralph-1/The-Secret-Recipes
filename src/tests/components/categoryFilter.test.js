import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CategoryFilter from '../../components/CategoriesFilter';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });
const setup = () => {
  const component = shallow(<CategoryFilter value="test" onChange={() => 'testing with jest'} />);
  return component;
};

describe('Header Component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<CategoryFilter value="test" onChange={() => 'testing with jest'} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  let component;
  beforeEach(() => {
    component = setup();
  });

  it('should have one span', () => {
    expect(component.find('select').length).toBe(1);
  });
});
