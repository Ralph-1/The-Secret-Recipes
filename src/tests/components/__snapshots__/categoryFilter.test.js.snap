// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches the snapshot 1`] = `
<div
  className="flex m-3 cat-filter"
>
  <h2>
    Select Category:
  </h2>
  <select
    className="filter"
    id="filter"
    onChange={[Function]}
    value="test"
  >
    <option
      value="Beef"
    >
      Beef
    </option>
    <option
      value="Chicken"
    >
      Chicken
    </option>
    <option
      value="Dessert"
    >
      Dessert
    </option>
    <option
      value="Lamb"
    >
      Lamb
    </option>
    <option
      value="Miscellaneous"
    >
      Miscellaneous
    </option>
    <option
      value="Pasta"
    >
      Pasta
    </option>
    <option
      value="Pork"
    >
      Pork
    </option>
    <option
      value="Seafood"
    >
      Seafood
    </option>
    <option
      value="Side"
    >
      Side
    </option>
    <option
      value="Starter"
    >
      Starter
    </option>
    <option
      value="Vegan"
    >
      Vegan
    </option>
    <option
      value="Vegetarian"
    >
      Vegetarian
    </option>
    <option
      value="Breakfast"
    >
      Breakfast
    </option>
    <option
      value="Goat"
    >
      Goat
    </option>
  </select>
</div>
`;
